samples/dml.js

```diff
@@ -630,6 +630,74 @@ async function updateUsingBatchDml(instanceId, databaseId, projectId) {
   // [END spanner_dml_batch_update]
 }
 
+async function executeSqlWithCustomTimeoutAndRetrySettings(
```

Comment:
Does the Samples Test fail due to the `async` prefix here? I don't see what the difference is, comparing to https://github.com/googleapis/nodejs-spanner/blob/0a50621ec24082ba0e4ccf932d5cb7402941092c/samples/dml.js#L20-L69.

Reply:
It's not about `async`, currently I am investigating on it.

Reply:
This issue turns up purely when you set the gaxOptions. If you don't, the sample runs fine.

Also, you can't set a subset of the fields. You have to set all of them.
 - first I set only the retry codes then it said `ERROR: TypeError: Cannot read property 'retryDelayMultiplier' of undefined`.
- then I set `retryDelayMultiplier` then it said `Error: 2 UNKNOWN: Getting metadata from plugin failed with error: Deadline is too far in the future`
- until you set each field, it keeps producing the same error.
- once you set all of them, it errors with `ERROR: TypeError: stream.on is not a function`.

So it has something to do with the fact that this is a streaming RPC. We set gaxOptions in the admin RPC calls in the system tests and this doesn't happen.

Reply:
@skuruppu @hengfengli  something with streaming in `google-gax` , I did check this in `bigtable` and found same error with stream.

Reply:
I see, we do not support retry in stream, [Here](https://github.com/googleapis/gax-nodejs/issues/729) is the issue in `google-gax`.